eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -1056,9 +1056,29 @@ private void updateDelta(Applications delta) {
      *             on any error.
      */
     private ClientResponse makeRemoteCall(Action action) throws Throwable {
+        if (lastRedirect.get() != null && isQueryAction(action)) {
+            String serviceUrl = lastRedirect.get();
```

Comment:
Minor race here, as lastRedirect.get() may change in between the two calls. Can cache it locally first by String serviceUrl = lastRedirect.get() before the if statement.